es-space-mgmt/main/default/lwc/reservationList/__tests__/reservationList.test.js

```diff
@@ -0,0 +1,231 @@
+    registerTestWireAdapter
+} from '@salesforce/sfdx-lwc-jest';
+import ReservationList from 'c/reservationList';
+import TILE_SELECTION_MC from '@salesforce/messageChannel/Tile_Selection__c';
+import FLOW_STATUS_CHANGE_MC from '@salesforce/messageChannel/Flow_Status_Change__c';
+import { subscribe, MessageContext, publish } from 'lightning/messageService';
+import { refreshApex } from '@salesforce/apex';
+
+// Realistic data with a list of spaces
+const mockOpenReservationsRecords = require('./data/getOpenReservations.json');
+// Register as Apex wire adapter. Some tests verify that provisioned values trigger desired behavior.
+const getOpenReservationsAdapter = registerApexTestWireAdapter(
+    getOpenReservations
+);
+// Register as a standard wire adapter because the component under test requires this adapter.
+// We don't exercise this wire adapter in the tests.
+const messageContextWireAdapter = registerTestWireAdapter(MessageContext);
+
+// mock apex refresh method
+jest.mock(
+    '@salesforce/apex',
+    () => {
+        return {
+            refreshApex: jest.fn()
+        };
+    },
+    { virtual: true }
+);
+
+describe('c-reservation-list', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+    });
+
+    it('getOpenReservations @wire data', () => {
+        const element = createElement('c-reservation-list', {
+            is: ReservationList
+        });
+        document.body.appendChild(element);
+
+        // Validate if pubsub got registered after connected to the DOM
+        expect(subscribe).toHaveBeenCalled();
+        expect(subscribe.mock.calls[0][1]).toBe(FLOW_STATUS_CHANGE_MC);
+
+        getOpenReservationsAdapter.emit(mockOpenReservationsRecords);
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const reservationTileElements = element.shadowRoot.querySelectorAll(
+                'c-reservation-tile'
+            );
+            expect(reservationTileElements.length).toBe(
+                mockOpenReservationsRecords.length
+            );
+            // check if the text is rendered fine
+            const openReservationParagraphElement = element.shadowRoot.querySelector(
+                'p.slds-text-heading_small'
+            );
+            expect(openReservationParagraphElement.textContent).toBe(
+                'Open Reservations:'
+            );
+        });
+    });
+
+    it('subscribe to flow status change message', () => {
```

Comment:
You're testing more than just subscribe in this test.
I would rename it "refreshes reservations when LMS message is received" (this assumes that we have subscribed first).

Reply:
done